hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -694,6 +697,32 @@ void prepareComputation(Bytes input, final UnaryOperator<byte[]> aliasResolver)
                                             encoder,
                                             evmEncoder,
                                             precompilePricingUtils));
+                            case AbiConstants.ABI_ID_HRC_ASSOCIATE -> checkHRC(
```

Comment:
Not for this PR, just throwing out ideas for the future but instead of a loooooonnng switch maybe we implement handlers that implement a single interface then HTSPrecompiledContract gets injected with all the singleton handlers and enum maps to the right handler and executes.
This way all the logic and some stuff from other places is contained in a class that speak to the specific system contract function